### PR TITLE
#252 test getSingleProject endpoint 

### LINKED ISSUE #252: Backend - Test getSingleProject Endpoint
### Description

Write unit tests for the getSingleProject endpoint. To work on this ticket, you will need to read through the endpoint to learn how it works.

### Acceptance Criteria

The following cases should be covered:
* the given wbs number is not a valid project wbs
* the associated wbselement does not exist
* it works as intended

### Proposed Solution

Check out this epic https://github.com/Northeastern-Electric-Racing/FinishLine/issues/238 for resources and tips on how to write these tests.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by a-geng**
## Changes
Tests for getSingleProject endpoint

## Test Cases
- fails given invalid project wbs
- fails when associated webselement doesnt exist
- works

Closes #252

### CODE REVIEW

File: src/backend/tests/test-data/projects.test-data.ts [hunk 1]
```diff
@@ -33,3 +33,37 @@ export const someProject = {
     ]
   }
 };
+
+export const plz = {
```
Comment (glacierA) [inline src/backend/tests/test-data/projects.test-data.ts:37]: change this lol

File: src/frontend/package.json [hunk 2]
```diff
@@ -56,12 +56,12 @@
   },
   "devDependencies": {
     "@testing-library/react-hooks": "^8.0.1",
-    "@types/react-bootstrap-table-next": "^4.0.18",
-    "@types/react-query": "^1.2.9",
```
Comment (glacierA) [inline src/frontend/package.json:60]: don't commit package.json

File: src/backend/tests/projects.test.ts [hunk 3]
```diff
@@ -119,4 +126,33 @@ describe('Projects', () => {
...
+    let res = await request(app).get('/1.0.1');
+    expect(res.statusCode).toBe(404);
+    expect(res.body).toStrictEqual({ message: `1.0.1 is not a valid project WBS #!` });
+
+    res = await request(app).get('/2.0.2');
+    expect(res.statusCode).toBe(404);
+    expect(res.body).toStrictEqual({ message: `2.0.2 is not a valid project WBS #!` });
+  });
+
+  test('getSingleProject fails when associated webselement doesnt exist', async () => {
```
Comment (glacierA) [inline src/backend/tests/projects.test.ts:140]: webselement?

**Review by glacierA — CHANGES_REQUESTED** [review]
nice job! rename a few things then we can merge it in.

**Review by glacierA — APPROVED** [review]
lgtm

File: src/backend/tests/projects.test.ts [hunk 4]
```diff
@@ -119,4 +126,34 @@ describe('Projects', () => {
...
+    expect(res.body).toStrictEqual({ message: 'project 1.3.0 not found!' });
+
+    res = await request(app).get('/2.4.0');
+    expect(res.statusCode).toBe(404);
+    expect(res.body).toStrictEqual({ message: 'project 2.4.0 not found!' });
+  });
+
+  test('getSingleProject works', async () => {
+    jest.spyOn(prisma.wBS_Element, 'findUnique').mockResolvedValue(someProject);
+    mockProjectTransformer.mockReturnValue({ message: 'projectTransformer called' });
```
Comment (anthonybernardi) [inline src/backend/tests/projects.test.ts:153]: clever!

**Review by anthonybernardi — APPROVED** [review]
nice job!

**Pull request merged**